ide/coqide.ml

```diff
@@ -110,7 +110,13 @@ let make_coqtop_args fname =
     | None -> args
     | Some fname ->
       if List.exists (String.equal "-top") args then args
-      else "-topfile"::fname::args
+      else
+        (* We basically copy the code of Names.check_valid since it is not exported *)
```

Comment:
Why not move this check to a library that can be shared between Coq and CoqIDE?

Alternatively, why not pass this name through the XML protocol rather than the command line, and have it validated by Coq?

Reply:
> Why not move this check to a library that can be shared between Coq and CoqIDE?

This is what I tried first (relying on `library_of_file` which `stm.ml` uses) but this would require at worst to be dependent on `loadpath.ml`, so I renounced.

> Alternatively, why not pass this name through the XML protocol rather than the command line, and have it validated by Coq?

This is probably an alternative. I don't know well the XML protocol. Can we add arbitrary new requests? (I'm also a bit afraid of the time it would take.)